test_manager/features/steps/tck_step_implementations.py

```diff
@@ -356,6 +368,23 @@ def set_key_to_ue_uri(context: Context, key: str, ue: str):
         context.json_dict[key] = context.ue_tracker[int(ue_number) - 1][2]
 
 
+@given('sets to entity URI of "{ue}" with updated "{sub_key}" to "{sub_val}"')
+@when('sets to entity URI of "{ue}" with updated "{sub_key}" to "{sub_val}"')
+def set_to_ue_uri(context: Context, ue: str, sub_key: str, sub_val: str):
+    ue_number = ue.replace("uE", "")
+    context.json_dict.update(context.ue_tracker[int(ue_number) - 1][2])
```

Comment:
ue_number? why index 2?

Reply:
Index 2 is the transport URI 

```
        context.ue_tracker.append(
            (
                test_agent_name,
                context.config.userdata[f"transport{str(counter)}"],
                current_uri,
                False,
            )
        )
```

Reply:
This is a dict which keep ue information. We need to know what transport URI are assigned to each uE so we can update uri for L1 and L2 api calls. 

The way this dict is setup index2 is where uri assigned to uE lives. 
This function pulls that uri and updated key value provided by the feature file